### PR TITLE
EEP 4: Type Annotation Integration

### PR DESCRIPTION
**Pull request by corranwebster**
Draft of EEP for type annotation integration via stub files.

### CODE REVIEW

File: docs/source/eeps/eep-4.rst [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+        value = Int()
+
+means that value holds an integer value, rather than an instance of :class:`Int`,
+without effectively declaring that :class:`Int` is a sub-type of :class:`int`,
+which can then lead to further issues.
+
+As a result, a more conservative approach is warranted, at least initially.
+Rather than starting to annotate directly and completely, it makes more sense
+to only partically annotate, and to do so with the use of stub files, so that
```
Comment (RobinD42) [inline docs/source/eeps/eep-4.rst]: partially (typo)

File: docs/source/eeps/eep-4.rst [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+  of specially-names methods.
+
+
+Implementation
+==============
+
+A reasonably initial step may be to use stub generation tools to generate
+stub files for most of the modules.  However most of the work will need to
+be done with the :module:`traits.trait_types` module, since those need to
+be converted into something that represents the
```
Comment (RobinD42) [inline docs/source/eeps/eep-4.rst]: represents the what?

File: docs/source/eeps/eep-4.rst [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+            items: bool = ...,
+            **metadata: _Dict[str, _Any]
+        ) -> None:
+            ...
+
+However some further overloading will be needed to cover the full signature
+of the class.
+
+With some work, we can hope to cover perhaps 80-90% of typical usage without
+giving warnings in cases where.
```
Comment (RobinD42) [inline docs/source/eeps/eep-4.rst]: in cases where what?

**Comment by mdickinson** [comment]
We failed to follow the process here: the proposed changes are already released in Traits.

@corranwebster Should we just merge this PR? Do you want to address the comments first?

**Pull request merged**